lnd_test.go

```diff
@@ -5407,11 +5546,118 @@ func testBasicChannelCreation(net *lntest.NetworkHarness, t *harnessTest) {
+	// active notification for each open channel.
+	var numChannelUpds int
+	for numChannelUpds < numChannels {
+		select {
+		case activeChannelUpdate := <-bobChanSub.updateChan:
+			if activeChannelUpdate.Type != lnrpc.ChannelUpdate_ACTIVE_CHANNEL {
+				t.Fatalf("update type mismatch: expected %v, got %v",
+					lnrpc.ChannelUpdate_ACTIVE_CHANNEL,
+					activeChannelUpdate.Type)
+			}
+
+			switch chanUpdate := activeChannelUpdate.Channel.(type) {
+			case *lnrpc.ChannelUpdate_OpenChannel:
+				if !chanUpdate.OpenChannel.Active {
+					t.Fatalf("channel not active but should be")
+				}
+			default:
+				t.Fatalf("channel update channel of wrong type, "+
+					"expected OpenChannel, got: %T",
+					chanUpdate)
+			}
+
+			numChannelUpds++
+		case <-time.After(time.Second * 10):
+			t.Fatalf("timeout waiting for channel notifications, "+
+				"only received %d/%d chanupds", numChannelUpds,
+				numChannels)
+		}
+	}
+
+	// Subscribe Alice to channel updates so we can test that both remote
+	// and local force close notifications are received correctly.
+	aliceChanSub := subscribeChannelNotifications(ctxb, t, net.Alice)
+	defer close(aliceChanSub.quit)
+
+	// Close the channel between Alice and Bob, asserting that the channel
+	// has been properly closed on-chain.
+	for i, chanPoint := range chanPoints {
 		ctx, _ := context.WithTimeout(context.Background(), timeout)
-		closeChannelAndAssert(ctx, t, net, net.Alice, chanPoint, false)
+
+		// Force close half of the channels.
+		force := i%2 == 0
+		closeChannelAndAssert(ctx, t, net, net.Alice, chanPoint, force)
+	}
+
+	errChan := make(chan error)
+	defer close(errChan)
+	var wg sync.WaitGroup
+
+	// verifyCloseUpdatesReceived is used to verify that Alice and Bob
+	// receive the correct channel updates in order.
+	verifyCloseUpdatesReceived := func(sub channelSubscription,
+		forceType lnrpc.ChannelCloseSummary_ClosureType) {
+
+		defer wg.Done()
+
+		// Ensure two notifications are received for each closed channel.
+		numChannelUpds := 0
+		for numChannelUpds < 2*numChannels {
+			// Every other channel should be force closed.
+			force := (numChannelUpds/2)%2 == 0
+			// Inactive ntfns should be received before close ntfns
+			// for a given channel.
+			inactiveNtfn := numChannelUpds%2 == 0
+
+			select {
+			case chanUpdate := <-sub.updateChan:
+				err := verifyCloseUpdate(chanUpdate, inactiveNtfn,
+					force, forceType)
+				if err != nil {
+					errChan <- err
+					return
+				}
+
+				numChannelUpds++
+			case err := <-sub.errChan:
+				errChan <- err
+				return
+			case <-time.After(time.Second * 10):
+				errChan <- fmt.Errorf("timeout waiting for channel "+
+					"notifications, only received %d/%d "+
+					"chanupds", numChannelUpds, 2*numChannels)
+				return
+			}
+		}
+	}
+
+	// Verify Bob receives all closed channel notifications. He should
+	// receive a remote force close notification for force closed channels.
+	wg.Add(1)
+	go verifyCloseUpdatesReceived(bobChanSub,
```

Comment:
What is the reason you are not checking this sequentially? If for runtime reasons, how much longer does the sequential test run?

Reply:
It only made a difference of 0.03 seconds actually, so I got rid of it.